### PR TITLE
Make DjangoDebugContext wait for nested fields

### LINKED ISSUE #293: __debug, __sql doesn't log sql queries made withing promises or resolves In Types
Hello world, I have a query that has promises with data loaders, In those dataloaders I do queries that don't show up in 
the queries 
```
__debug{ __sql{ sql}}
```

### PR DESCRIPTION
**Pull request by ktosiek**
This commit makes DjangoDebugContext wait for all field's promises, even for fields that only started their resolvers after __debug was resolved.

Fixes #293.

### CODE REVIEW

**Comment by japrogramer** [comment]
why hasn't the checks completed for this pull request?

**Comment by hypercodex** [comment]
What needs to happen for this to get merged?

**Comment by zbyte64** [comment]
@ktosiek For whatever reason travis CI never built this PR. Please merge with master and push to re-trigger the build.

**Comment by ktosiek** [comment]
@zbyte64 thank you for the suggestion, it looks like the tests are passing now

File: graphene_django/debug/middleware.py [hunk 1]
```diff
@@ -16,14 +16,18 @@ def __init__(self):
     def get_debug_promise(self):
         if not self.debug_promise:
             self.debug_promise = Promise.all(self.promises)
+            self.promises = []
```
[the remark this answers is not in the document]
Comment (ktosiek) [reply graphene_django/debug/middleware.py]: In on_resolve_all_promises I'm checking this list to see if there are any pending promises. If it's empty I know the debug_promise really is the last one, and nothing interesting happened concurrently while resolving the debug_promise.

**Pull request merged**